### PR TITLE
system variable: add tidb_opt_skew_distinct_agg

### PR DESCRIPTION
**Pull request by fixdb**
system variable: add `tidb_opt_skew_distinct_agg`

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]
LGTM

File: system-variables.md [hunk 1]
```diff
@@ -1544,6 +1544,12 @@ explain select * from t where age=5;
 3 rows in set (0.00 sec)
 ```
 
+### `tidb_opt_skew_distinct_agg` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
```
Comment (TomShawn) [inline system-variables.md:1549]: 是否持久化到集群呢？

File: system-variables.md [hunk 2]
```diff
@@ -1544,6 +1544,12 @@ explain select * from t where age=5;
...
+- 默认值：`OFF`
+- 这个变量用来设置优化器是否将带有 `distinct` 的聚合函数（比如 `select b, count(distinct a) from t group by b`）改写为两层聚合函数（`select b, count(a) from (select b, a from t group by b, a) t group by b`）。当聚合列有严重的数据倾斜，且distinct列有很多不同的值时，这种改写能够避免查询执行过程中的数据倾斜，达到性能提升的效果。
```
Comment (TomShawn) [inline system-variables.md:1551]:
```suggestion
- 这个变量用来设置优化器是否将带有 `distinct` 的聚合函数（例如 `select b, count(distinct a) from t group by b`）改写为两层聚合函数（例如 `select b, count(a) from (select b, a from t group by b, a) t group by b`）。当聚合列有严重的数据倾斜，且 `distinct` 列有很多不同的值时，这种改写能够避免查询执行过程中的数据倾斜，从而提升查询性能。
```

Comment (TomShawn) [reply system-variables.md:1551]: 另外，能否将 SQL 语句的关键字大写呢？

Comment (fixdb) [reply system-variables.md]: done

File: system-variables.md [hunk 3]
```diff
@@ -1544,6 +1544,12 @@ explain select * from t where age=5;
 3 rows in set (0.00 sec)
 ```
 
+### `tidb_opt_skew_distinct_agg` <span class="version-mark">从 v6.1.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:1547]:
```suggestion
### `tidb_opt_skew_distinct_agg` <span class="version-mark">从 v6.2.0 版本开始引入</span>
```

File: system-variables.md (hunk 1 above)
Comment (TomShawn) [reply system-variables.md:1549]:
```suggestion
- 作用域：SESSION | GLOBAL
- 是否持久化到集群：否
```

File: system-variables.md [hunk 4]
```diff
@@ -1544,6 +1544,13 @@ explain select * from t where age=5;
 3 rows in set (0.00 sec)
 ```
 
+### `tidb_opt_skew_distinct_agg` <span class="version-mark">从 v6.2.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：否
+- 默认值：`OFF`
+- 这个变量用来设置优化器是否将带有 `DISTINCT` 的聚合函数（例如 `SELECT b, count(DISTINCT a) FROM t GROUP BY b`）改写为两层聚合函数（例如 `SELECT b, count(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`）。当聚合列有严重的数据倾斜，且 `DISTINCT` 列有很多不同的值时，这种改写能够避免查询执行过程中的数据倾斜，从而提升查询性能。此优化仅对TiFlash有效。
```
Comment (TomShawn) [inline system-variables.md:1552]:
```suggestion
- 这个变量用来设置优化器是否将带有 `DISTINCT` 的聚合函数（例如 `SELECT b, count(DISTINCT a) FROM t GROUP BY b`）改写为两层聚合函数（例如 `SELECT b, count(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`）。当聚合列有严重的数据倾斜，且 `DISTINCT` 列有很多不同的值时，这种改写能够避免查询执行过程中的数据倾斜，从而提升查询性能。此优化**仅对 TiFlash 有效**。
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: system-variables.md [hunk 5]
```diff
@@ -1544,6 +1544,13 @@ explain select * from t where age=5;
 3 rows in set (0.00 sec)
 ```
 
+### `tidb_opt_skew_distinct_agg` <span class="version-mark">从 v6.2.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：否
```
Comment (seiya-annie) [inline system-variables.md:1550]: need change to : yes

**Pull request merged**